client/proxy-dbus.cc

```diff
@@ -0,0 +1,360 @@
+/*
+ * Copyright (c) 2016 SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact Novell, Inc.
+ *
+ * To contact Novell about this file by physical or electronic mail, you may
+ * find current contact information at www.novell.com.
+ */
+
+
+#include "proxy-dbus.h"
+#include "commands.h"
+#include "utils/text.h"
+
+
+using namespace std;
+
+
+ProxySnapshotDbus::ProxySnapshotDbus(ProxySnapshotsDbus* backref, unsigned int num)
+    : backref(backref), num(num)
+{
+    XSnapshot x = command_get_xsnapshot(conn(), config_name(), num);
+
+    type = x.type;
+    date = x.date;
+    uid = x.uid;
+    pre_num = x.pre_num;
+    description = x.description;
+    cleanup = x.cleanup;
+    userdata = x.userdata;
+}
+
+
+ProxySnapshotDbus::ProxySnapshotDbus(ProxySnapshotsDbus* backref, SnapshotType type, unsigned int num,
+				     time_t date, uid_t uid, unsigned int pre_num,
+				     const string& description, const string& cleanup,
+				     const map<string, string>& userdata)
+    : backref(backref), type(type), num(num), date(date), uid(uid), pre_num(pre_num),
```

Comment:
If you write that one element per line, it's very easy to compare it against the data members in the header file. Like this, that's very tedious.